robosat/osm/road.py

```diff
@@ -0,0 +1,160 @@
+        },
+        'secondary': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.75
+        },
+        'tertiary': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.75
+        },
+        'unclassified': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.00
+        },
+        'residential': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.75
+        },
+        'service': { 
+            'lanes': 1,
+            'lane_width': 3.00,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.00
+        },
+        'motorway_link': { 
+            'lanes': 2,
+            'lane_width': 3.75,
+            'left_hard_shoulder_width': 0.75,
+            'right_hard_shoulder_width': 3.00
+        },
+        'trunk_link': { 
+            'lanes': 2,
+            'lane_width': 3.75,
+            'left_hard_shoulder_width': 0.50,
+            'right_hard_shoulder_width': 1.50
+        },
+        'primary_link': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.75
+        },
+        'secondary_link': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.75
+        },
+        'tertiary_link': { 
+            'lanes': 1,
+            'lane_width': 3.50,
+            'left_hard_shoulder_width': 0.00,
+            'right_hard_shoulder_width': 0.00
+        }
+    }
+
+    EARTH_MEAN_RADIUS = 6371004.0
+
+    def __init__(self):
+        super().__init__()
+        self.features = []
+
+    def way(self, w):
+        if "highway" not in w.tags:
+            return
+
+        if w.tags["highway"] not in self.road_filter():
+            return
+
+        left_hard_shoulder_width = self.highway_attributes[w.tags["highway"]]['left_hard_shoulder_width']
+        lane_width = self.highway_attributes[w.tags["highway"]]['lane_width']
+        lanes = self.highway_attributes[w.tags["highway"]]['lanes']
+        right_hard_shoulder_width = self.highway_attributes[w.tags["highway"]]['right_hard_shoulder_width']
+
+        if "oneway" not in w.tags:
+            lanes = lanes * 2
+        elif w.tags["oneway"] == "no":
+            lanes = lanes * 2
+
+        if "lanes" in w.tags:
+            lanes = int(w.tags["lanes"])
+
+        road_width = left_hard_shoulder_width + lane_width * lanes + right_hard_shoulder_width
+
+        if "width" in w.tags:
+            road_width = float(w.tags["width"])
```

Comment:
The `float()` conversion might fail

https://taginfo.openstreetmap.org/keys/width#values

check the values; some folks don't just add floats. We probably should try-catch here.